File: .net & mono/Weed3/Base/IQuery.cs
using Noear.Weed.Cache;
using System;
using System.Collections.Generic;

namespace Noear.Weed {
    public interface IQuery {
        long getCount();
        T getValue<T>(T def);
        T getValue<T>(T def, Action<CacheUsing, T> cacheCondition);
        T getItem<T>() where T : IBinder;
        T getItem<T>(Action<CacheUsing, T> cacheCondition) where T : IBinder;
        List<T> getList<T>() where T : IBinder;
        List<T> getList<T>(Action<CacheUsing, List<T>> cacheCondition) where T : IBinder;
        List<T> getArray<T>(String column);
        DataList getDataList();
        DataList getDataList(Action<CacheUsing, DataList> cacheCondition);
        DataItem getDataItem();
        DataItem getDataItem(Action<CacheUsing, DataList> cacheCondition);

        IQuery caching(ICacheService service);
        IQuery usingCache(bool isCache);
        IQuery usingCache(int seconds);
        IQuery cacheTag(String tag);
    }
}

File: java/Weed3/src/noear/weed/IQuery.java
package noear.weed;

import noear.weed.cache.CacheUsing;
import noear.weed.cache.ICacheService;
import noear.weed.ext.Act2;

import java.sql.SQLException;
import java.util.List;

/**
 * Created by yuety on 14/11/12.
 */
public interface IQuery {
    public long getCount() throws SQLException;
    public Object getValue() throws SQLException;
    public <T> T getValue(T def) throws SQLException;
    public <T> T getValue(T def, Act2<CacheUsing, T> cacheCondition) throws SQLException;
    public <T extends IBinder> T getItem(T model) throws SQLException;
    public <T extends IBinder> T getItem(T model, Act2<CacheUsing, T> cacheCondition) throws SQLException;
    public <T extends IBinder> List<T> getList(T model) throws SQLException;
    public <T extends IBinder> List<T> getList(T model, Act2<CacheUsing, List<T>> cacheCondition) throws SQLException;
    public DataList getDataList() throws SQLException;
    public DataList getDataList(Act2<CacheUsing, DataList> cacheCondition) throws SQLException;
    public DataItem getDataItem() throws SQLException;
    public DataItem getDataItem(Act2<CacheUsing, DataList> cacheCondition) throws SQLException;

    public IQuery caching(ICacheService service);
    public IQuery usingCache(boolean isCache);
    public IQuery usingCache(int seconds);
    public IQuery cacheTag(String tag);
}
